src/rpc/server/scala/serv/src/main/scala/com/ctc/omega_edit/grpc/EditorService.scala

```diff
@@ -459,6 +460,36 @@ class EditorService(implicit val system: ActorSystem) extends Editor {
             SegmentResponse.of(in.sessionId, offset, ByteString.copyFrom(data))
           )
       }
+
+  def stopServer(kind: ServerControlKind): ServerControlResponse =
+    Await.ready(system.terminate(), 1.minute).value.get match {
+      case Success(_) => ServerControlResponse(kind, 0)
+      case Failure(_) => ServerControlResponse(kind, 1)
+    }
+
+  def serverControl(in: ServerControlRequest): Future[ServerControlResponse] =
+    in.kind match {
+      case ServerControlKind.SERVER_CONTROL_GRACEFUL_SHUTDOWN => {
+        // TODO: Implement graceful shutdown. Wait till all sessions are. destroyed then shutdown.
+
+        // Future.successful(stopServer(in.kind))
+        Future.failed(
+          grpcFailure(Status.UNKNOWN, s"${in.kind} NEEDS IMPLEMENTED")
+        )
+      }
+      case ServerControlKind.SERVER_CONTROL_IMMEDIATE_SHUTDOWN => {
+        DestroyActors()
```

Comment:
What do you think is happening here?

Reply:
So the thought was for destroying all existing sessions, which are akka actors right? So it loops through all `context.children` and destroys each actor (session).
Is it doing something different than anticipated?

Reply:
Ok. I thought it looks like DestroyActors is instantiated but but nothing is done with it.